stages/org.osbuild.ostree.deploy

```diff
@@ -105,44 +105,54 @@ def make_fs_identifier(desc):
     raise ValueError("unknown rootfs type")
 
 
-def main(tree, inputs, options):
-    osname = options["osname"]
-    rootfs = options.get("rootfs")
-    mounts = options.get("mounts", [])
-    kopts = options.get("kernel_opts", [])
-    ref = options["ref"]
-    remote = options.get("remote")
+def ostree_commit_deploy(tree, inputs, osname, remote, ref, kopts):
+    if len(inputs) == 0:
+        if not ref:
+            raise ValueError("ref should be specified in options")
+    elif len(inputs) == 1:
+        if ref:
+            raise ValueError("Should not specify ref if input was specified")
 
-    # If provided an input then do the pull into the tree
-    if len(inputs) != 0:
+        # If we have an input then we need to pull_local() from the input
+        # first before we deploy.
         source_repo, commits = ostree.parse_input_commits(inputs["commits"])
         target_repo = f"{tree}/ostree/repo"
         for commit, data in commits.items():
-            loopref = data.get("ref", commit)
-            ostree.pull_local(source_repo, target_repo, remote, loopref)
+            ref = data.get("ref", commit)
+            ostree.pull_local(source_repo, target_repo, remote, ref)
 
     if remote:
         ref = f"{remote}:{ref}"
```

Comment:
Will setting the remote work if `ref` is a commit ID read from the source commit metadata?

Reply:
yes. it works. here is what it looks like:

```
[core@cosa-devsh ~]$ rpm-ostree status 
State: idle
AutomaticUpdatesDriver: Zincati
  DriverState: active; trying to stage 38.20231002.3.1 (failed attempts: 1)
Deployments:
● fedora:2c39e9b6a6b26226b626f4daf815d5307a96b2cd57463e94f7c618e312a61980
                  Version: 38.20230902.3.0 (2023-09-18T22:16:48Z)
                   Commit: 2c39e9b6a6b26226b626f4daf815d5307a96b2cd57463e94f7c618e312a61980
             GPGSignature: Valid signature by 6A51BBABBA3D5467B6171221809A8D7CEB10B464
```

We could definitely change things to still accept `ref` as input even if it was also specified in the input commit and then use that instead of the input commit ref if the user specifies it.

Reply:
That sounds like a good idea.  Can do in a followup.